packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/template/ConvenienceSyncMethodTemplate.java

```diff
@@ -95,10 +95,16 @@ protected void writeInvocationAndConversion(ClientMethod convenienceMethod, Clie
             = convenienceMethod.getType() == ClientMethodType.SimpleSyncRestResponse ? "" : ".getValue()";
 
         if (convenienceMethod.getType() == ClientMethodType.PagingSync) {
-            methodBlock.methodReturn(String.format("serviceClient.%1$s(%2$s).mapPage(bodyItemValue -> %3$s)",
-                protocolMethod.getName(), invocationExpression,
-                expressionConvertFromBinaryData(responseBodyType, rawResponseBodyType, "bodyItemValue",
-                    protocolMethod.getProxyMethod().getResponseContentTypes(), typeReferenceStaticClasses)));
+            String mapPageExpression = "";
+            if (JavaSettings.getInstance().isBranded()) {
+                mapPageExpression
+                    = ".mapPage(bodyItemValue -> "
+                        + expressionConvertFromBinaryData(responseBodyType, rawResponseBodyType, "bodyItemValue",
+                            protocolMethod.getProxyMethod().getResponseContentTypes(), typeReferenceStaticClasses)
+                        + ")";
+            }
+            methodBlock.methodReturn(String.format("serviceClient.%1$s(%2$s)%3$s", protocolMethod.getName(),
+                invocationExpression, mapPageExpression));
```

Comment:
This be the main fix. We should not use `mapPage` in unbranded, as this is not available in clientcore, and we already generate `PagedIterable<Model>` for unbranded, therefore no need to convert the type.